From c336ab3050f9b38634d2bfb80e4d416be1954609 Mon Sep 17 00:00:00 2001
From: Praneeth Bajjuri <praneeth@ti.com>
Date: Fri, 11 Mar 2011 18:22:52 -0600
Subject: OMAP4: VDAC override for untrimmed units

This patch is to do software override for VDAC to
the trim value as per recomendation for some ES2.2 silicon revs

this is needed to avoid instabilities at low OPPs

Change-Id: I0262e9fac97e2c19ca727ea608e4d10f774194b9
Signed-off-by: Praneeth Bajjuri <praneeth@ti.com>
---
 arch/arm/mach-omap2/voltage.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/voltage.c b/arch/arm/mach-omap2/voltage.c
index df61fe5..12076d4 100644
--- a/arch/arm/mach-omap2/voltage.c
+++ b/arch/arm/mach-omap2/voltage.c
@@ -2480,9 +2480,10 @@ static int __init omap_voltage_init(void)
 	/*
 	 * Some ES2.2 efuse  values for BGAP and SLDO trim
 	 * are not programmed. For these units
-	 * we can set overide mode for SLDO trim,
+	 * 1. we can set overide mode for SLDO trim,
 	 * and program the max multiplication factor, to ensure
 	 * high enough voltage on SLDO output.
+	 * 2. trim VDAC value for TV output as per recomendation
 	 */
 	if (cpu_is_omap44xx()
 		&& (omap_rev() == OMAP4430_REV_ES2_2)) {
@@ -2498,6 +2499,9 @@ static int __init omap_voltage_init(void)
 			OMAP4_CTRL_MODULE_CORE_LDOSRAM_CORE_VOLTAGE_CTRL);
 			omap_ctrl_writel(0x0401040f,
 			OMAP4_CTRL_MODULE_CORE_LDOSRAM_IVA_VOLTAGE_CTRL);
+			/* write value of 0x0 to VDAC as per trim recomendation */
+			omap_ctrl_writel(0x000001c0,
+			OMAP4_CTRL_MODULE_PAD_CORE_CONTROL_EFUSE_1);
 		}
 	}
 #ifdef CONFIG_PM_DEBUG
-- 
1.6.0.4

